tools/ci/testlist/arm-05.dat

```diff
@@ -1,4 +1,17 @@
-/arm/[m-q]*,CONFIG_ARM_TOOLCHAIN_GNU_EABI
```

Comment:
let's simply switch this row to clang? since clang build is little, it's always good to switch more gcc to clangw.

Reply:
If all are changed to clang, many other logs are as follows:
```
  Configuration/Tool: mps3-an547/picostest,CONFIG_ARM_TOOLCHAIN_CLANG
  2024-10-28 17:25:26
  ------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Disabling CONFIG_ARM_TOOLCHAIN_GNU_EABI
  Enabling CONFIG_ARM_TOOLCHAIN_CLANG
```
This is a non-clang compilation configuration, but it enables CONFIG_ARM_TOOLCHAIN_CLANG when compiling testbuilg.sh. Although the final compilation result is passed, it should be the dependency that caused it to fail to be enabled.Is this a bug?
```
function configure_default {
  if ! ./tools/configure.sh ${HOPTION} ${STORE} $config ${JOPTION} 1>/dev/null; then
    fail=1
  fi

  if [ "X$toolchain" != "X" ]; then
    setting=`grep _TOOLCHAIN_ $nuttx/.config | grep -v CONFIG_ARCH_TOOLCHAIN_* | grep =y`
    original_toolchain=`echo $setting | cut -d'=' -f1`
    if [ ! -z "$original_toolchain" ]; then
      echo "  Disabling $original_toolchain"
      kconfig-tweak --file $nuttx/.config -d $original_toolchain
    fi

    echo "  Enabling $toolchain"
    kconfig-tweak --file $nuttx/.config -e $toolchain

    makefunc olddefconfig
  fi

  return $fail
}
```
If the other defconfig is GNU, then it will be disabled according to the writing of testbuild.sh. So if xxx.dat is changed to the default clang, then it will be tried to enable it all the time.